code/game/machinery/computer/_computer.dm

```diff
@@ -28,6 +30,21 @@
 	. = ..()
 	power_change()
 
+/obj/machinery/computer/CanAllowThrough(atom/movable/mover, border_dir) // allows projectiles to fly over the computer
+	. = ..()
+	if(!projectiles_pass_chance)
+		return FALSE
+	if(!isprojectile(mover))
+		return FALSE
+	var/obj/projectile/proj = mover
+	if(!anchored)
+		return TRUE
+	if(proj.firer && Adjacent(proj.firer) && projectiles_pass_chance) // projectiles_pass_chance check is here in case you don't want a computer to act like a barricade
+		return TRUE
```

Comment:
You're already checking for !projectiles_pass_chance earlier

Reply:
fixed